components/eamxx/src/physics/ml_correction/eamxx_ml_correction_process_interface.cpp

```diff
@@ -135,6 +147,61 @@ void MLCorrection::run_impl(const double dt) {
       ML_model_tq, ML_model_uv, ML_model_sfc_fluxes, datetime_str);
   pybind11::gil_scoped_release no_gil;  
   ekat::enable_fpes(fpe_mask);   
+
+  // Now back out the qv change abd apply it to precipitation, only if Tq ML is turned on
+  if (not (m_ML_model_path_tq == "None")) {
+    using PC  = scream::physics::Constants<Real>;
+    using KT  = KokkosTypes<DefaultDevice>;
+    using MT  = typename KT::MemberType;
+    using ESU = ekat::ExeSpaceUtils<typename KT::ExeSpace>;
+    const auto &pseudo_density       = get_field_in("pseudo_density").get_view<const Real**>();
+    const auto &precip_liq_surf_mass = get_field_out("precip_liq_surf_mass").get_view<Real *, Host>();
+    const auto &precip_ice_surf_mass = get_field_out("precip_ice_surf_mass").get_view<Real *, Host>();
+    constexpr Real g = PC::gravit;
+    const auto num_levs = m_num_levs;
+    const auto policy = ESU::get_default_team_policy(m_num_cols, m_num_levs);
+    
+    const auto &qv_tnew = get_field_in("qv").get_view<const Real **>();
+    Kokkos::parallel_for("Compute WVP diff", policy,
+                         KOKKOS_LAMBDA(const MT& team) {
+      const int icol = team.league_rank();
+      auto qold_icol = ekat::subview(qv_told,icol);
+      auto qnew_icol = ekat::subview(qv_tnew,icol);
+      auto rho_icol  = ekat::subview(pseudo_density,icol);
+      Real wvp = 0;
+      // Compute WaterVaporPath Difference
+      Kokkos::parallel_reduce(Kokkos::TeamVectorRange(team, num_levs),
+                              [&] (const int& ilev, Real& lsum) {
+        lsum += (qnew_icol(ilev)-qold_icol(ilev)) * rho_icol(ilev) / g;
+      },wvp);
+      team.team_barrier();
+      // Adjust Precipitation
+      //  - Note, we subtract the water vapor path because positive precip represents
+      //    a descrease in qv.
+      auto tot_precip = precip_liq_surf_mass(icol)+precip_ice_surf_mass(icol);
+      if (tot_precip>0) {
+        // adjust precip by weighted avg of both phases
+        auto liq_w = precip_liq_surf_mass(icol)/tot_precip;
+        auto ice_w = precip_ice_surf_mass(icol)/tot_precip;
+        precip_liq_surf_mass(icol) -= liq_w*wvp;
```

Comment:
This is not thread safe. I know you are running this only on Host, for now, but we should avoid issues down the road, if ML becomes device friendly. You should wrap all these blocks in a `Kokkos::single` (with `Kokkos::PerTeam`.